acceptance/lib/puppet/acceptance/install_utils.rb

```diff
@@ -120,65 +120,53 @@ def stop_firewall_on(host)
+        project         = sha == 'nightly' ? project + '-latest'                :  project
+        sha             = sha == 'nightly' ? nil                                :  sha
+
+        if sha == 'nightly'
+          case platform
+          when /^(fedora|el|centos)-(\d+)-(.+)$/
+            variant = (($1 == 'centos') ? 'el' : $1)
+            fedora_prefix = ((variant == 'fedora') ? 'f' : '')
+            version = $2
+            arch = $3
+
+            repo_filename = "pl-%s%s-%s-%s%s-%s.repo" % [
+              project,
+              sha ? '-' + sha : '',
+              variant,
+              fedora_prefix,
+              version,
+              arch
+            ]
+            repo_url = "%s/%s/%s/repo_configs/rpm/%s" % [tld, project, sha, repo_filename]
+
+            on host, "curl -o /etc/yum.repos.d/#{repo_filename} #{repo_url}"
+          when /^(debian|ubuntu|cumulus)-([^-]+)-(.+)$/
+            variant = $1
+            version = $2
+            arch = $3
+
+            if variant =~ /cumulus/ then
+              version = variant
+            end
 
-          list_filename = "pl-%s%s-%s.list" % [
-            project,
-            sha ? '-' + sha : '',
-            version
-          ]
-          list_url = "http://%s/%s/%s/repo_configs/deb/%s" % [tld, project, sha, list_filename]
+            list_filename = "pl-%s%s-%s.list" % [
+              project,
+              sha ? '-' + sha : '',
+              version
+            ]
+            list_url = "%s/%s/%s/repo_configs/deb/%s" % [tld, project, sha, list_filename]
 
-          on host, "curl -o /etc/apt/sources.list.d/#{list_filename} #{list_url}"
-          on host, "apt-get update"
-        else
-          if project == 'puppet-agent'
-            opts = {
-              :puppet_collection => 'PC1',
-              :puppet_agent_sha => ENV['SHA'],
-              # SUITE_VERSION is necessary for Beaker to build a package download
-              # url which is built upon a `git describe` for a SHA.
-              # Beaker currently cannot find or calculate this value based on
-              # the SHA, and thus it must be passed at invocation time.
-              # The one exception is when SHA is a tag like `1.8.0` and
-              # SUITE_VERSION will be equivalent.
-              # RE-8333 may make this unnecessary in the future
-              :puppet_agent_version => ENV['SUITE_VERSION'] || ENV['SHA']
-            }
-            # this installs puppet-agent on windows (msi), osx (dmg) and eos (swix)
-            install_puppet_agent_dev_repo_on(agent, opts)
+            on host, "curl -o /etc/apt/sources.list.d/#{list_filename} #{list_url}"
+            on host, "apt-get update"
           else
             fail_test("No repository installation step for #{platform} yet...")
           end
+        else
+          install_from_build_data_url(project, "#{tld}/#{project}/#{sha}/artifacts/#{sha}.yaml", host)
```

Comment:
@johnduarte I decided you were right, and it was silly to overwrite `sha` with `ENV['SHA']` if I didn't have to. I've kicked off a new run of the ad hoc pipeline to make sure this works as expected.